guides/cppguide.xml

```diff
@@ -3054,68 +3054,69 @@ Thomas Voss
       </p>
       <ul>
-        <li> The return type is always on the same line as the
-             function name.
-             </li>
+        <li> <code>void</code> or <code>auto</code>  is always on the same line
+             as the function name.
+        </li>
         <li> The open parenthesis is always on the same line as the
              function name.
-             </li>
+        </li>
         <li> There is never a space between the function name and the
              open parenthesis.
-             </li>
+        </li>
         <li> There is never a space between the parentheses and the
              parameters.
-             </li>
-        <li> The open curly brace is always on the line following the last parameter.
-             </li>
+        </li>
+        <li> The open curly brace is always on the line following the last parameter
+             or return type.
+       </li>
         <li> The close curly brace is either on the last line by itself
              or (if other style rules permit) on the same line as the
              open curly brace.
-             </li>
+       </li>
         <li> All parameters should be named, with identical names in the
              declaration and implementation. (Except where unused parameters
              are suppressed in the implementation.)
-             </li>
+       </li>
         <li> All parameters should be in a single line if possible, otherwise:
             <ul>
-                <li> Wrap groups of parameters into the next line, and align the 
-                     first wrapped parameter to the first parameter in the previous 
-                     line.
-                     </li>
-                <li> Place each parameter in a separate line and indent each with 4 
-                     spaces.
-                     </li>
+              <li> Place each parameter in a separate line and indent each with 4
+                spaces.
+              </li>
+              <li> Wrap groups of parameters into the next line, and align the
+                   first wrapped parameter to the first parameter in the previous
+                   line.
+              </li>
             </ul>
-             </li>
+        </li>
       </ul>
       <p>
         If your function is <code>const</code>, the <code>const</code>
         keyword should be on the same line as the last parameter:
       </p>
       <CODE_SNIPPET>
         // Everything in this function signature fits on a single line
-        ReturnType function_name(Type par) const
+        auto function_name(Type par) const -> ReturnType
         {
             ...
         }
 
         // This function signature requires multiple lines, but
         // the const keyword is on the line with the last parameter.
-        ReturnType really_long_function_name(Type par1,
-                                             Type par2) const
+        void really_long_function_name(Type par1,
+                                       Type par2) const
```

Comment:
Could you kindly fix the indentation of the function arguments while you're here?